files/build_templates/updategraph.service.j2

```diff
@@ -6,6 +6,10 @@ Requires=database.service
 
 [Service]
 Type=oneshot
+{% if sonic_asic_platform == 'mellanox' -%}
+EnvironmentFile=/host/machine.conf
+ExecStartPre=/bin/bash -c "/usr/share/sonic/device/$onie_platform/hw-management start"
```

Comment:
>hw-management [](start = 66, length = 13)

Is updategraph a good location for hw-management?

Reply:
Agree with Qi. I don't see the reason to put hw-management here, and I don't see the need to change this into a j2 file either. Platform information can also be tested during runtime.

Reply:
we need drivers to access eeprom
j2 to avoid this code for others

Reply:
Why not create another service for this?
And, you can create a shell script to enable hw-management in which you can test platform information run time. It is much easier to follow than a build-time j2 rendering.

Reply:
in current case, when will hw-management be started? Maybe hw-management itself should be a service and make it start early.